src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx

```diff
@@ -0,0 +1,109 @@
+import { renderHook, waitFor } from '../../../test-utils/testing-library';
+import messages from '../../common/messages';
+import { FIELD_PERMISSIONS_CAN_UPLOAD } from '../../../constants';
+import useSidebarMetadataFetcher, { STATUS } from '../hooks/useSidebarMetadataFetcher';
+
+const mockError = {
+    status: 500,
+    message: 'Internal Server Error',
+};
+
+const mockFile = {
+    id: '123',
+    permissions: { [FIELD_PERMISSIONS_CAN_UPLOAD]: true },
+};
+
+const mockTemplates = [
+    {
+        id: 'metadata_template_custom_1',
+        scope: 'global',
+        templateKey: 'properties',
+        hidden: false,
+    },
+];
+
+const mockAPI = {
+    getFile: jest.fn((id, successCallback, errorCallback) => {
+        try {
+            successCallback(mockFile);
+        } catch (error) {
+            errorCallback(error);
+        }
+    }),
+    getMetadata: jest.fn((_file, successCallback, errorCallback) => {
+        try {
+            successCallback({
+                editors: [],
+                templates: mockTemplates,
+            });
+        } catch (error) {
+            errorCallback(error);
+        }
+    }),
+};
+const api = {
+    getFileAPI: jest.fn().mockReturnValue(mockAPI),
+    getMetadataAPI: jest.fn().mockReturnValue(mockAPI),
+};
+
+describe('useSidebarMetadataFetcher', () => {
+    const onErrorMock = jest.fn();
+    const isFeatureEnabledMock = true;
+
+    const setupHook = (fileId = '123') =>
+        renderHook(() => useSidebarMetadataFetcher(api, fileId, onErrorMock, isFeatureEnabledMock));
+
+    it('should fetch the file and metadata successfully', async () => {
```

Comment:
```suggestion
    test('should fetch the file and metadata successfully', async () => {
```
all the `it` test suites should be `test`

Reply:
sure, changed